tests/sources/test_zoom.py

```diff
@@ -0,0 +1,805 @@
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+FILE_WITHOUT_EXTENSION = {
+    "file_id": "file2",
+    "type": "file",
+    "date_time": "2023-02-09T00:00:00Z",
+    "file_size": 300,
+    "file_name": "file4",
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+FILE_WITH_LARGE_DATA = {
+    "file_id": "file3",
+    "type": "file",
+    "date_time": "2023-02-09T00:00:00Z",
+    "file_size": 10485761,
+    "file_name": "file5.txt",
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+FILE_WITH_UNSUPPORTED_EXTENSION = {
+    "file_id": "file4",
+    "type": "file",
+    "date_time": "2023-02-09T00:00:00Z",
+    "file_size": 300,
+    "file_name": "file3.png",
+    "download_url": "https://api.zoom.us/v2/download_url",
+}
+SAMPLE_FILE = {
+    "next_page_token": None,
+    "messages": [
+        FILE,
+        FILE_WITHOUT_EXTENSION,
+        FILE_WITH_LARGE_DATA,
+        FILE_WITH_UNSUPPORTED_EXTENSION,
+    ],
+}
+EXPECTED_FILES_RESPONSE = [
+    {
+        "file_id": "file1",
+        "type": "file",
+        "date_time": "2023-03-09T00:00:00Z",
+        "file_size": 100,
+        "file_name": "file1.txt",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file1",
+        "_id": "file1",
+        "_timestamp": "2023-03-09T00:00:00Z",
+    },
+    {
+        "file_id": "file2",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 300,
+        "file_name": "file4",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file2",
+        "_id": "file2",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+    {
+        "file_id": "file3",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 10485761,
+        "file_name": "file5.txt",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file3",
+        "_id": "file3",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+    {
+        "file_id": "file4",
+        "type": "file",
+        "date_time": "2023-02-09T00:00:00Z",
+        "file_size": 300,
+        "file_name": "file3.png",
+        "download_url": "https://api.zoom.us/v2/download_url",
+        "id": "file4",
+        "_id": "file4",
+        "_timestamp": "2023-02-09T00:00:00Z",
+    },
+]
+
+
+class ZoomAsyncMock(mock.AsyncMock):
+    def __init__(self, data, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._data = data
+
+    async def json(self):
+        return self._data
+
+    async def text(self):
+        return self._data
+
+
+def get_mock(mock_response):
+    async_mock = mock.AsyncMock()
+    async_mock.__aenter__ = mock.AsyncMock(
+        return_value=ZoomAsyncMock(data=mock_response)
+    )
+    return async_mock
+
+
+def setup_zoom(source, fetch_past_meeting_details=False):
+    # Set up default config with default values
+    source.configuration.set_field(name="account_id", value="123")
```

Comment:
We're moving away from using `set_field` in tests as it doesn't update a Field, it creates an entirely new one. This loses all of the configuration data set by the default configuration.

A better way to change configuration is `source.configuration.get_field("foo").value = "bar"`

Reply:
Actually, I notice that you're calling `setup_zoom` for almost every test.

You could streamline this by initialising the `create_source` with a wrapper like `create_zoom_source` and then pass in the configuration changes immediately. This would mean the source won't initialise with empty data.

I've done this in the Salesforce test files, [example](https://github.com/elastic/connectors-python/blob/52e4c25bd85b11029d63bdb941068141710ccf08/tests/sources/test_salesforce.py#L423-L445).

Reply:
Forget to mention this is follow up pr, Suggested changes are made in #1504 and on this [commit](https://github.com/elastic/connectors-python/pull/1504/commits/9de198335c2e01befc718589c387ea33ee8a2379)

Reply:
Thanks for addressing it in the other PR. I still had a follow-up nit (but at the end of the day it's just a nit). Nice work!